---
title: "Find Bias Features - Single-nucleus RNA Sequencing Data"
package: "BiasDetect"
author:
  - name: "Christine Hou"
    affiliation: Department of Biostatistics, Johns Hopkins University
    email: chris2018hou@gmail.com
output: BiocStyle::html_document
vignette: >
  %\VignetteIndexEntry{01 Find Bias Features - Spatial Transcriptomics Data}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
    collapse = TRUE,
    comment = "#>"
)
```

### Introduction

`BiasDetect` package was developed on an unrelated DLPFC dataset available
through the 
[spatialLIBD](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-022-08601-w) 
package. We first chose **binomial deviance model** from
[scry](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1861-6) 
as the feature selection method that can incorporate a batch variable into the
model. Next, we compared the per-gene ranks and dispersion values when the 
model was run with and without a batch effect. Our data-driven thresholding
approach is using the cutoffs based on the number of standard deviance (nSD)
of deviance and rank difference metrics. Based on self-selected nSD cutoffs, 
the genes identified as biased can be filtered out.

The methodology details can be found in 
[Find Bias Feature](https://jac-thom.github.io/findBiasedFeatures/) written by
Jacqui Thompson, and the documentation related codes can be found on
[GitHub](https://github.com/jac-thom/findBiasedFeatures).

### Installation

`BiasDetect` is a R package. Install development version from
[GitHub](https://christinehou11.github.io/BiasDetect):

```{r 'install dev', eval = FALSE}
remotes::install("christinehou11/BiasDetect")
```

### Setup

Install `BiasDetect`

```{r, eval=FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE)) {
    install.packages("BiocManager")
}

BiocManager::install("BiasDetect")

## Check that you have a valid Bioconductor installation
BiocManager::valid()
```

Load required packages

```{r 'library', message=FALSE}
library("BiasDetect")
library("SummarizedExperiment")
library("SpatialExperiment")
```

### Data

```{r hidden data input,message=FALSE, warning=FALSE}

# snranseq
```

### Bias Genes Identification



### Additional information

Please read 
[reference documentation](https://github.com/jac-thom/findBiasedFeatures) for
detailed codes and scripts conducting PRECAST clustering
[before](https://github.com/jac-thom/findBiasedFeatures/blob/main/code/02_spatialHPC_PRECAST-svgs.R) 
and 
[after](https://github.com/jac-thom/findBiasedFeatures/blob/main/code/05_spatialHPC_PRECAST-svgs-no-bias.R) 
removing the biased gene.

### `R` session information {.unnumbered}

```{r 'sessionInfo'}
## Session info
sessionInfo()
```
