---
title: "Find Bias Features - Spatial Transcriptomics Data"
package: "BiasDetect"
author:
  - name: "Christine Hou"
    affiliation: Department of Biostatistics, Johns Hopkins University
    email: chris2018hou@gmail.com
  - name: "Jacqui Thompson"
    affiliation: Department of Biostatistics, Johns Hopkins University
    email: jthom338@jh.edu
  - name: "Stephanie Hicks"
    affiliation: Department of Biostatistics, Johns Hopkins University
    email: shicks19@jhu.edu
output: BiocStyle::html_document
vignette: >
  %\VignetteIndexEntry{01 Tutorial for spe data object}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
    collapse = TRUE,
    comment = "#>"
)
```

### Introduction

`BiasDetect` package was developed on an unrelated DLPFC dataset available through the [spatialLIBD](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-022-08601-w) package. We first chose **binomial deviance model** from [scry](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1861-6) as the feature selection method that can incorporate a batch variable into the model. Next, we compared the per-gene ranks and dispersion values when the model was run with and without a batch effect. Our data-driven thresholding approach is using the cutoffs based on the number of standard deviance (nSD) of deviance and rank difference metrics. Based on self-selected nSD cutoffs, the genes identified as biased can be filtered out.

The methodology details can be found in [Find Bias Feature](https://jac-thom.github.io/findBiasedFeatures/) written by Jacqui Thompson, and the documentation related codes can be found on [GitHub](https://github.com/jac-thom/findBiasedFeatures).

### Installation

`BiasDetect` is a R package. Install development version from [GitHub](https://christinehou11.github.io/BiasDetect).

```{r 'install dev', eval = FALSE}
remotes::install("christinehou11/BiasDetect")
```

### Setup

Install `BiasDetect`

```{r, eval=FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE)) {
    install.packages("BiocManager")
}

BiocManager::install("BiasDetect")

## Check that you have a valid Bioconductor installation
BiocManager::valid()
```

Load required packages

```{r 'library', message=FALSE}
library(BiasDetect)
library(SpatialExperiment)
library(dplyr)
library(tidyr)
library(tibble)
```

### Data

`humanHippocampus2024` is an `ExperimentHub` package. The package is still under review for Bioconductor community, but the development version can be can be installed from GitHub.

```{r github data package, eval = FALSE, comment=NA, message=FALSE, warning=FALSE}
install.packages("remotes")
remotes::install_github("christinehou11/humanHippocampus2024")
```

The package stored `spe` object used for this tutorial. The detailed tutorial to retrieve the data objects from `humanHippocampus2024` package is [here](https://christinehou11.github.io/humanHippocampus2024/articles/humanHippocampus2024.html).

```{r hidden data input,message=FALSE, warning=FALSE}
library(humanHippocampus2024)
library(ExperimentHub)

ehub <- ExperimentHub()
myfiles <- query(ehub, "humanHippocampus2024")
spatial_hpc_spe <- myfiles[["EH9605"]]
spatial_hpc_spe
```

The dataset in this tutorial section is from [spatialHPC](https://github.com/LieberInstitute/spatial_hpc). We filtered out 4 samples.

```{r spe filter,comment=NA, message=FALSE, warning=FALSE}
fix_order = distinct(as.data.frame(colData(spatial_hpc_spe)), slide, array, brnum, sample_id, position, sex) %>% 
  arrange(slide, array)
sub4 = fix_order$sample_id[c(14,16,
                             20,21)]
spe_sub4 = spatial_hpc_spe[,spatial_hpc_spe$sample_id %in% sub4]

```

We ran [nnSVG](https://www.nature.com/articles/s41467-023-39748-z) to identify spatially-variable genes (SVGs). 

```{r nnsvg, comment=NA, message=FALSE, warning=FALSE}
load(system.file("extdata","nnSVG_outs_HE_only.rda", package="BiasDetect"))
dim(res_ranks)
```

We filtered to only the top 2000 significant features in the 4 samples we're using and further filterer to only features that are in the top 2000 of >1 sample. Finally, we obtained 2082 SVGs.

```{r nnsvg filter, message=FALSE, warning=FALSE, comment=NA}
res_df = pivot_longer(
    rownames_to_column(as.data.frame(res_ranks),
                        var="gene_id"), 
    colnames(res_ranks), names_to="sample_id", 
    values_to="rank", values_drop_na=T)

res_df2 = filter(res_df, 
    sample_id %in% 
    c("V11L05-333_B1","V11L05-333_D1",
    "V11L05-335_D1", "V11L05-336_A1"), 
    rank<=2000)

svgs = group_by(res_df2, gene_id) %>% tally() %>% filter(n>1) 
nrow(svgs)

spe_sub4 = spe_sub4[rowData(spe_sub4)$gene_id %in% svgs$gene_id,]
dim(spe_sub4)
```


### Bias Genes Identification

The data frame created from `featureSelect()` function includes `nSD_dev` and `nSD_rank`, and we can use `biasDetect()` function to identify the bias genes based on the self-selected `nSD` integer thresholds for deviance and rank respectively. The default threshold is 5 for deviance and 5 for rank.

```{r feature select, comment = NA, message=FALSE, warning=FALSE}
SVGs <- rowData(spe_sub4)$gene_name
batch_df <- featureSelect(spe_sub4, batch_effect = "sample_id", VGs = SVGs)
dim(batch_df)
head(batch_df)
```

What's more, `biasDetect` function allows to return the biased genes either in data frame format or in plots. The default format is in data frame.

```{r bias genes identification, comment = NA, message=FALSE, warning=FALSE}
bias <- biasDetect(batch_df = batch_df, nSD_dev = 10, nSD_rank = 5)
bias
```

To see the plots, add `visual = TRUE`.

```{r visual, comment = NA, message=FALSE, warning=FALSE}
bias_fig <- biasDetect(batch_df = batch_df, nSD_dev = 10, nSD_rank = 5, visual = TRUE)
```

*Figure a) SVGs with relative change in deviance >= 10 SDs from the mean*

```{r dev, comment = NA, message=FALSE, warning=FALSE}
bias_fig$deviance
```

*Figure b) SVGs with rank difference >= 5 SDs from the mean*

```{r rank, comment = NA, message=FALSE, warning=FALSE}
bias_fig$rank
```

Finally, we can obtain a newer `spe_sub4` object without the biased genes.

```{r new spe_sub4, comment = NA, message=FALSE, warning=FALSE}
svgs_filt = setdiff(rownames(spe_sub4), bias)
length(svgs_filt)
```


### Additional information

Please read [reference documentation](https://github.com/jac-thom/findBiasedFeatures) for detailed codes and scripts conducting PRECAST clustering [before](https://github.com/jac-thom/findBiasedFeatures/blob/main/code/02_spatialHPC_PRECAST-svgs.R) and [after](https://github.com/jac-thom/findBiasedFeatures/blob/main/code/05_spatialHPC_PRECAST-svgs-no-bias.R) removing the biased gene.

### `R` session information {.unnumbered}

```{r 'sessionInfo'}
## Session info
sessionInfo()
```
