<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Feature Selection — featureSelect • BiasDetect</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Feature Selection — featureSelect"><meta name="description" content="Function to conduct feature selection on snRNA-seq and spatial
transcriptomics data and calculate the difference of deviance and rank
with and without the selected batch effect."><meta property="og:description" content="Function to conduct feature selection on snRNA-seq and spatial
transcriptomics data and calculate the difference of deviance and rank
with and without the selected batch effect."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">BiasDetect</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/snRNA-seq.html">Find Bias Features - Single-nucleus RNA Sequencing Data</a></li>
    <li><a class="dropdown-item" href="../articles/spe.html">Find Bias Features - Spatial Transcriptomics Data</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/christinehou11/BiasDetect/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Feature Selection</h1>
      <small class="dont-index">Source: <a href="https://github.com/christinehou11/BiasDetect/blob/main/R/featureSelect.R" class="external-link"><code>R/featureSelect.R</code></a></small>
      <div class="d-none name"><code>featureSelect.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Function to conduct feature selection on snRNA-seq and spatial
transcriptomics data and calculate the difference of deviance and rank
with and without the selected batch effect.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">featureSelect</span><span class="op">(</span><span class="va">input</span>, batch_effect <span class="op">=</span> <span class="cn">NULL</span>, VGs <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-input">input<a class="anchor" aria-label="anchor" href="#arg-input"></a></dt>
<dd><p><code>SpatialExperiment</code> or <code>SingleCellExperiment</code>: Input
data which can be either <code>SpatialExperiment</code> or
<code>SingleCellExperiment</code>. It is assumed to have an <code>assays</code> slot
containing <code>counts</code> assay for <code>devianceFeatureSelection()</code> to
successfully operate and calculate the deviance and rank. The
<code>logcounts</code> is strongly recommended to be included in <code>assays</code>.
Also, the input data is assumed to set up <code>rownames(input)</code> as
<code>genes</code> and <code>gene_name</code> in <code>rowData(input)</code>. The input can
be either the raw and complete data object or the filtered data object
containing only highly variable genes (HVGs) or spatially varaible genes
(SVGs).</p></dd>


<dt id="arg-batch-effect">batch_effect<a class="anchor" aria-label="anchor" href="#arg-batch-effect"></a></dt>
<dd><p><code>character</code>: Either batch on <code>slide</code> or
<code>subject</code> based on what metadata is available within input data.
The name of <code>slide</code> or <code>subject</code> within each input object
should be specified based on different scenarios.</p></dd>


<dt id="arg-vgs">VGs<a class="anchor" aria-label="anchor" href="#arg-vgs"></a></dt>
<dd><p><code>character</code>: Highly Variable Genes (HVGs) for
<code>SingleCelleExperiment</code> object or Spatially Variable Genes (SVGs)
for <code>SpatialExperiment</code> object. If it is a data frame, it is assumed
to contain one column of identified variable genes with column name as
"gene_name".</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>If the input was provided as a <code>SpatialExperiment</code> or
<code>SingleCellExperiment</code> object, the output values are returned as
a data frame containing the deviance and rank with and without the bacth
effect, the corresponding difference, the corresponding nSD, gene, gene
name, and whether the gene is outlier defined by the chosen deviance and
rank cutoff.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># dropbox link</span></span></span>
<span class="r-in"><span><span class="va">file_name</span> <span class="op">&lt;-</span> <span class="st">"https://www.dropbox.com/scl/fi/mppcfm2ays6qet7y1x46i/spe.rda"</span></span></span>
<span class="r-in"><span><span class="va">suffix</span> <span class="op">&lt;-</span> <span class="st">"?rlkey=1i138uo808pjrsv957hnfbtmf&amp;st=qod6xg3m&amp;dl=1"</span></span></span>
<span class="r-in"><span><span class="va">url</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">file_name</span>,<span class="va">suffix</span>,sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># load data</span></span></span>
<span class="r-in"><span><span class="va">response</span> <span class="op">&lt;-</span> <span class="fu">httr</span><span class="fu">::</span><span class="fu"><a href="https://httr.r-lib.org/reference/GET.html" class="external-link">GET</a></span><span class="op">(</span><span class="va">url</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rawConnection.html" class="external-link">rawConnection</a></span><span class="op">(</span><span class="va">response</span><span class="op">$</span><span class="va">content</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># conduct feature selection</span></span></span>
<span class="r-in"><span><span class="va">SVGs</span> <span class="op">&lt;-</span> <span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">$</span><span class="va">gene_name</span></span></span>
<span class="r-in"><span><span class="va">batch_df</span> <span class="op">&lt;-</span> <span class="fu">featureSelect</span><span class="op">(</span><span class="va">spe</span>, batch_effect <span class="op">=</span> <span class="st">"sample_id"</span>, VGs <span class="op">=</span> <span class="va">SVGs</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Step 1: Running feature selection without batch...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Step 2: Running feature selection with batch...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Step 3: Calculating deviance and rank difference...</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Christine Hou, Jacqui Thompson, Stephanie C. Hicks.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

