<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Find Bias Features - Single-nucleus RNA Sequencing Data â€¢ BiasDetect</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Find Bias Features - Single-nucleus RNA Sequencing Data">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">BiasDetect</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/snRNA-seq.html">Find Bias Features - Single-nucleus RNA Sequencing Data</a></li>
    <li><a class="dropdown-item" href="../articles/spe.html">Find Bias Features - Spatial Transcriptomics Data</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/christinehou11/BiasDetect/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Find Bias Features - Single-nucleus RNA Sequencing Data</h1>
                        <h4 data-toc-skip class="author">Christine
Hou</h4>
            <address class="author_afil">
      Department of Biostatistics, Johns Hopkins
University<br><a class="author_email" href="mailto:#"></a><a href="mailto:chris2018hou@gmail.com" class="email">chris2018hou@gmail.com</a>
      </address>
                              <h4 data-toc-skip class="author">Jacqui
Thompson</h4>
            <address class="author_afil">
      Department of Biostatistics, Johns Hopkins
University<br><a class="author_email" href="mailto:#"></a><a href="mailto:jthom338@jh.edu" class="email">jthom338@jh.edu</a>
      </address>
                              <h4 data-toc-skip class="author">Stephanie
Hicks</h4>
            <address class="author_afil">
      Department of Biostatistics, Johns Hopkins
University<br><a class="author_email" href="mailto:#"></a><a href="mailto:shicks19@jhu.edu" class="email">shicks19@jhu.edu</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/christinehou11/BiasDetect/blob/main/vignettes/snRNA-seq.Rmd" class="external-link"><code>vignettes/snRNA-seq.Rmd</code></a></small>
      <div class="d-none name"><code>snRNA-seq.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h3>
<p><code>BiasDetect</code> package was developed on an unrelated DLPFC
dataset available through the <a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-022-08601-w" class="external-link">spatialLIBD</a>
package. We first chose <strong>binomial deviance model</strong> from <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1861-6" class="external-link">scry</a>
as the feature selection method that can incorporate a batch variable
into the model. Next, we compared the per-gene ranks and dispersion
values when the model was run with and without a batch effect. Our
data-driven thresholding approach is using the cutoffs based on the
number of standard deviance (nSD) of deviance and rank difference
metrics. Based on self-selected nSD cutoffs, the genes identified as
biased can be filtered out.</p>
<p>The methodology details can be found in <a href="https://jac-thom.github.io/findBiasedFeatures/" class="external-link">Find Bias
Feature</a> written by Jacqui Thompson, and the documentation related
codes can be found on <a href="https://github.com/jac-thom/findBiasedFeatures" class="external-link">GitHub</a>.</p>
</div>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p><code>BiasDetect</code> is a R package. Install development version
from <a href="https://christinehou11.github.io/BiasDetect">GitHub</a>:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install</span><span class="op">(</span><span class="st">"christinehou11/BiasDetect"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h3>
<p>Install <code>BiasDetect</code></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"BiasDetect"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check that you have a valid Bioconductor installation</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/valid.html" class="external-link">valid</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Load required packages</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://christinehou11.github.io/BiasDetect/">"BiasDetect"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">"SummarizedExperiment"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/drighelli/SpatialExperiment" class="external-link">"SpatialExperiment"</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># snranseq</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="bias-genes-identification">Bias Genes Identification<a class="anchor" aria-label="anchor" href="#bias-genes-identification"></a>
</h3>
</div>
<div class="section level3">
<h3 id="additional-information">Additional information<a class="anchor" aria-label="anchor" href="#additional-information"></a>
</h3>
<p>Please read <a href="https://github.com/jac-thom/findBiasedFeatures" class="external-link">reference
documentation</a> for detailed codes and scripts conducting PRECAST
clustering <a href="https://github.com/jac-thom/findBiasedFeatures/blob/main/code/02_spatialHPC_PRECAST-svgs.R" class="external-link">before</a>
and <a href="https://github.com/jac-thom/findBiasedFeatures/blob/main/code/05_spatialHPC_PRECAST-svgs-no-bias.R" class="external-link">after</a>
removing the biased gene.</p>
</div>
<div class="section level3 unnumbered">
<h3 class="unnumbered" id="r-session-information">
<code>R</code> session information<a class="anchor" aria-label="anchor" href="#r-session-information"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Session info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.5 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] SpatialExperiment_1.14.0    SingleCellExperiment_1.26.0</span></span>
<span><span class="co">#&gt;  [3] SummarizedExperiment_1.34.0 Biobase_2.64.0             </span></span>
<span><span class="co">#&gt;  [5] GenomicRanges_1.56.2        GenomeInfoDb_1.40.1        </span></span>
<span><span class="co">#&gt;  [7] IRanges_2.38.1              S4Vectors_0.42.1           </span></span>
<span><span class="co">#&gt;  [9] BiocGenerics_0.50.0         MatrixGenerics_1.16.0      </span></span>
<span><span class="co">#&gt; [11] matrixStats_1.4.1           BiasDetect_0.99.0          </span></span>
<span><span class="co">#&gt; [13] BiocStyle_2.32.1           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] gtable_0.3.5            rjson_0.2.23            xfun_0.48              </span></span>
<span><span class="co">#&gt;  [4] bslib_0.8.0             ggplot2_3.5.1           ggrepel_0.9.6          </span></span>
<span><span class="co">#&gt;  [7] lattice_0.22-6          vctrs_0.6.5             tools_4.4.1            </span></span>
<span><span class="co">#&gt; [10] generics_0.1.3          parallel_4.4.1          tibble_3.2.1           </span></span>
<span><span class="co">#&gt; [13] fansi_1.0.6             pkgconfig_2.0.3         Matrix_1.7-0           </span></span>
<span><span class="co">#&gt; [16] RColorBrewer_1.1-3      desc_1.4.3              lifecycle_1.0.4        </span></span>
<span><span class="co">#&gt; [19] GenomeInfoDbData_1.2.12 compiler_4.4.1          textshaping_0.4.0      </span></span>
<span><span class="co">#&gt; [22] munsell_0.5.1           codetools_0.2-20        htmltools_0.5.8.1      </span></span>
<span><span class="co">#&gt; [25] sass_0.4.9              yaml_2.3.10             pkgdown_2.1.1          </span></span>
<span><span class="co">#&gt; [28] pillar_1.9.0            crayon_1.5.3            jquerylib_0.1.4        </span></span>
<span><span class="co">#&gt; [31] BiocParallel_1.38.0     DelayedArray_0.30.1     cachem_1.1.0           </span></span>
<span><span class="co">#&gt; [34] magick_2.8.5            abind_1.4-8             rsvd_1.0.5             </span></span>
<span><span class="co">#&gt; [37] tidyselect_1.2.1        digest_0.6.37           BiocSingular_1.20.0    </span></span>
<span><span class="co">#&gt; [40] dplyr_1.1.4             bookdown_0.41           fastmap_1.2.0          </span></span>
<span><span class="co">#&gt; [43] grid_4.4.1              colorspace_2.1-1        cli_3.6.3              </span></span>
<span><span class="co">#&gt; [46] SparseArray_1.4.8       magrittr_2.0.3          S4Arrays_1.4.1         </span></span>
<span><span class="co">#&gt; [49] utf8_1.2.4              UCSC.utils_1.0.0        scales_1.3.0           </span></span>
<span><span class="co">#&gt; [52] rmarkdown_2.28          XVector_0.44.0          httr_1.4.7             </span></span>
<span><span class="co">#&gt; [55] ragg_1.3.3              beachmat_2.20.0         ScaledMatrix_1.12.0    </span></span>
<span><span class="co">#&gt; [58] evaluate_1.0.1          knitr_1.48              irlba_2.3.5.1          </span></span>
<span><span class="co">#&gt; [61] rlang_1.1.4             Rcpp_1.0.13             glue_1.8.0             </span></span>
<span><span class="co">#&gt; [64] BiocManager_1.30.25     scry_1.16.0             jsonlite_1.8.9         </span></span>
<span><span class="co">#&gt; [67] R6_2.5.1                systemfonts_1.1.0       fs_1.6.4               </span></span>
<span><span class="co">#&gt; [70] zlibbioc_1.50.0</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Christine Hou, Jacqui Thompson, Stephanie C. Hicks.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
