<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Find Bias Features - Spatial Transcriptomics Data • BiasDetect</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Find Bias Features - Spatial Transcriptomics Data">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">BiasDetect</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/snRNA-seq.html">Find Bias Features - Single-nucleus RNA Sequencing Data</a></li>
    <li><a class="dropdown-item" href="../articles/spe.html">Find Bias Features - Spatial Transcriptomics Data</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/christinehou11/BiasDetect/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Find Bias Features - Spatial Transcriptomics Data</h1>
                        <h4 data-toc-skip class="author">Christine
Hou</h4>
            <address class="author_afil">
      Department of Biostatistics, Johns Hopkins
University<br><a class="author_email" href="mailto:#"></a><a href="mailto:chris2018hou@gmail.com" class="email">chris2018hou@gmail.com</a>
      </address>
                              <h4 data-toc-skip class="author">Jacqui
Thompson</h4>
            <address class="author_afil">
      Department of Biostatistics, Johns Hopkins
University<br><a class="author_email" href="mailto:#"></a><a href="mailto:jthom338@jh.edu" class="email">jthom338@jh.edu</a>
      </address>
                              <h4 data-toc-skip class="author">Stephanie
Hicks</h4>
            <address class="author_afil">
      Department of Biostatistics, Johns Hopkins
University<br><a class="author_email" href="mailto:#"></a><a href="mailto:shicks19@jhu.edu" class="email">shicks19@jhu.edu</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/christinehou11/BiasDetect/blob/main/vignettes/spe.Rmd" class="external-link"><code>vignettes/spe.Rmd</code></a></small>
      <div class="d-none name"><code>spe.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h3>
<p><code>BiasDetect</code> package was developed on an unrelated DLPFC
dataset available through the <a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-022-08601-w" class="external-link">spatialLIBD</a>
package. We first chose <strong>binomial deviance model</strong> from <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1861-6" class="external-link">scry</a>
as the feature selection method that can incorporate a batch variable
into the model. Next, we compared the per-gene ranks and dispersion
values when the model was run with and without a batch effect. Our
data-driven thresholding approach is using the cutoffs based on the
number of standard deviance (nSD) of deviance and rank difference
metrics. Based on self-selected nSD cutoffs, the genes identified as
biased can be filtered out.</p>
<p>The methodology details can be found in <a href="https://jac-thom.github.io/findBiasedFeatures/" class="external-link">Find Bias
Feature</a> written by Jacqui Thompson, and the documentation related
codes can be found on <a href="https://github.com/jac-thom/findBiasedFeatures" class="external-link">GitHub</a>.</p>
</div>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p><code>BiasDetect</code> is a R package. Install development version
from <a href="https://christinehou11.github.io/BiasDetect">GitHub</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install</span><span class="op">(</span><span class="st">"christinehou11/BiasDetect"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h3>
<p>Install <code>BiasDetect</code></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"BiasDetect"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check that you have a valid Bioconductor installation</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/valid.html" class="external-link">valid</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Load required packages</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://christinehou11.github.io/BiasDetect/">BiasDetect</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/drighelli/SpatialExperiment" class="external-link">SpatialExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://httr.r-lib.org/" class="external-link">httr</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a>
</h3>
<p>The <code>spe</code> object used for this tutorial is stored in <a href="https://www.dropbox.com/home/biasDetect_vignette?preview=spe.rda" class="external-link">Dropbox</a>.</p>
<p>The object is also saved in the <code>ExperimentHub</code> package
named <code>humanHippocampus2024</code>, and the development version
from <a href="(https://github.com/christinehou11/humanHippocampus2024)" class="external-link">GitHub</a>
is available. The package has not been publicly available, but the data
objects can be accessed via GitHub installation and tutorial is <a href="https://christinehou11.github.io/humanHippocampus2024/" class="external-link">here</a>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">github</span> <span class="op">&lt;-</span> <span class="st">"https://github.com/christinehou11/BiasDetect-analyses/"</span></span>
<span><span class="va">subdir</span> <span class="op">&lt;-</span> <span class="st">"raw/refs/heads/main/processed-data/spatialHPC_SRT/"</span></span>
<span><span class="va">csv1</span> <span class="op">&lt;-</span> <span class="st">"spe-hpc_sub4_svgs-only_counts-1.csv"</span></span>
<span><span class="va">csv2</span> <span class="op">&lt;-</span> <span class="st">"spe-hpc_sub4_svgs-only_counts-2.csv"</span></span>
<span><span class="va">cdata</span> <span class="op">&lt;-</span> <span class="st">"spe-hpc_sub4_svgs-only_colData.csv"</span></span>
<span><span class="va">rdata</span> <span class="op">&lt;-</span> <span class="st">"spe-hpc_sub4_svgs-only_rowData.csv"</span></span>
<span></span>
<span><span class="va">url1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">github</span>,<span class="va">subdir</span>,<span class="va">csv1</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span>
<span><span class="va">url2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">github</span>,<span class="va">subdir</span>,<span class="va">csv2</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span>
<span><span class="va">url3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">github</span>,<span class="va">subdir</span>,<span class="va">cdata</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span>
<span><span class="va">url4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">github</span>,<span class="va">subdir</span>,<span class="va">rdata</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span>
<span></span>
<span><span class="va">file1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="va">url1</span><span class="op">)</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">file2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="va">url2</span><span class="op">)</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">file1</span>, <span class="va">file2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\."</span>,<span class="st">"-"</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">cdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="va">url3</span><span class="op">)</span>, row.names<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">rdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="va">url4</span><span class="op">)</span>, row.names<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment.html" class="external-link">SpatialExperiment</a></span><span class="op">(</span></span>
<span>    assay <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"counts"</span><span class="op">=</span><span class="va">counts</span><span class="op">)</span>, </span>
<span>    colData <span class="op">=</span> <span class="va">cdata</span>, rowData <span class="op">=</span> <span class="va">rdata</span>,</span>
<span>    spatialCoordsNames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"array_row"</span>, <span class="st">"array_col"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>The dataset in this tutorial section is from <a href="https://github.com/LieberInstitute/spatial_hpc" class="external-link">spatialHPC</a>. We
filtered out 4 samples. We then ran <a href="https://www.nature.com/articles/s41467-023-39748-z" class="external-link">nnSVG</a> to
identify spatially-variable genes (SVGs). We filtered to only the top
2000 significant features in the 4 samples we’re using and further
filterer to only features that are in the top 2000 of &gt;1 sample.
Finally, we obtained 2082 SVGs.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>spe</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>class<span class="sc">:</span> SpatialExperiment </span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>dim<span class="sc">:</span> <span class="dv">2082</span> <span class="dv">18945</span> </span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="fu">metadata</span>(<span class="dv">0</span>)<span class="sc">:</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="fu">assays</span>(<span class="dv">1</span>)<span class="sc">:</span> counts</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="fu">rownames</span>(<span class="dv">2082</span>)<span class="sc">:</span> ENSG00000002330 ENSG00000002586 ... ENSG00000286214</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>  ENSG00000286961</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>rowData <span class="fu">names</span>(<span class="dv">6</span>)<span class="sc">:</span> source type ... gene_name gene_type</span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="fu">colnames</span>(<span class="dv">18945</span>)<span class="sc">:</span> AAACAACGAATAGTTC<span class="dv">-1</span>_V11L05<span class="dv">-333</span>_B1</span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>  AAACAAGTATCTCCCA<span class="dv">-1</span>_V11L05<span class="dv">-333</span>_B1 ... TTGTTTGTATTACACG<span class="dv">-1</span>_V11L05<span class="dv">-336</span>_A1</span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>  TTGTTTGTGTAAATTC<span class="dv">-1</span>_V11L05<span class="dv">-336</span>_A1</span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a>colData <span class="fu">names</span>(<span class="dv">10</span>)<span class="sc">:</span> sample_id slide ... expr_chrM expr_chrM_ratio</span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a><span class="fu">reducedDimNames</span>(<span class="dv">0</span>)<span class="sc">:</span></span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a>mainExpName<span class="sc">:</span> <span class="cn">NULL</span></span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a><span class="fu">altExpNames</span>(<span class="dv">0</span>)<span class="sc">:</span></span>
<span id="cb5-16"><a href="#cb5-16" tabindex="-1"></a>spatialCoords <span class="fu">names</span>(<span class="dv">2</span>) <span class="sc">:</span> array_row array_col</span>
<span id="cb5-17"><a href="#cb5-17" tabindex="-1"></a>imgData <span class="fu">names</span>(<span class="dv">0</span>)<span class="sc">:</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="bias-genes-identification">Bias Genes Identification<a class="anchor" aria-label="anchor" href="#bias-genes-identification"></a>
</h3>
<p>The data frame created from <code><a href="../reference/featureSelect.html">featureSelect()</a></code> function
includes <code>nSD_dev</code> and <code>nSD_rank</code>, and we can use
<code><a href="../reference/biasDetect.html">biasDetect()</a></code> function to identify the bias genes based on
the self-selected <code>nSD</code> integer thresholds for deviance and
rank respectively. The default threshold is 5 for deviance and 5 for
rank.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>SVGs <span class="ot">&lt;-</span> <span class="fu">rowData</span>(spe)<span class="sc">$</span>gene_id</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>batch_df <span class="ot">&lt;-</span> <span class="fu">featureSelect</span>(spe, <span class="at">batch_effect =</span> <span class="st">"sample_id"</span>, <span class="at">VGs =</span> SVGs)</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>Step <span class="dv">1</span><span class="sc">:</span> Running feature selection without batch...</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>Step <span class="dv">2</span><span class="sc">:</span> Running feature selection with batch...</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>Step <span class="dv">3</span><span class="sc">:</span> Calculating deviance and rank difference...</span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="fu">dim</span>(batch_df)</span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>[<span class="dv">1</span>] <span class="dv">2082</span>   <span class="dv">10</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="fu">head</span>(batch_df)</span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a>             gene gene_name dev_default rank_default dev_batch rank_batch</span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a><span class="dv">1</span> ENSG00000002330       BAD    <span class="fl">15830.43</span>         <span class="dv">1310</span>  <span class="fl">15785.12</span>       <span class="dv">1289</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="dv">2</span> ENSG00000002586      CD99    <span class="fl">18837.21</span>          <span class="dv">865</span>  <span class="fl">18616.10</span>        <span class="dv">858</span></span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a><span class="dv">3</span> ENSG00000004059      ARF5    <span class="fl">15033.32</span>         <span class="dv">1443</span>  <span class="fl">14985.57</span>       <span class="dv">1417</span></span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a><span class="dv">4</span> ENSG00000004660    CAMKK1    <span class="fl">14586.54</span>         <span class="dv">1506</span>  <span class="fl">14479.27</span>       <span class="dv">1487</span></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a><span class="dv">5</span> ENSG00000004779   NDUFAB1    <span class="fl">19859.78</span>          <span class="dv">756</span>  <span class="fl">19469.72</span>        <span class="dv">764</span></span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a><span class="dv">6</span> ENSG00000005022   SLC25A5    <span class="fl">16842.03</span>         <span class="dv">1150</span>  <span class="fl">16739.61</span>       <span class="dv">1131</span></span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a>       d_diff     nSD_dev r_diff   nSD_rank</span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a><span class="dv">1</span> <span class="fl">0.002870288</span> <span class="sc">-</span><span class="fl">0.31052071</span>    <span class="sc">-</span><span class="dv">21</span> <span class="sc">-</span><span class="fl">0.4914075</span></span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a><span class="dv">2</span> <span class="fl">0.011877487</span> <span class="sc">-</span><span class="fl">0.13869673</span>     <span class="sc">-</span><span class="dv">7</span> <span class="sc">-</span><span class="fl">0.1638025</span></span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a><span class="dv">3</span> <span class="fl">0.003186151</span> <span class="sc">-</span><span class="fl">0.30449522</span>    <span class="sc">-</span><span class="dv">26</span> <span class="sc">-</span><span class="fl">0.6084093</span></span>
<span id="cb6-20"><a href="#cb6-20" tabindex="-1"></a><span class="dv">4</span> <span class="fl">0.007408683</span> <span class="sc">-</span><span class="fl">0.22394495</span>    <span class="sc">-</span><span class="dv">19</span> <span class="sc">-</span><span class="fl">0.4446068</span></span>
<span id="cb6-21"><a href="#cb6-21" tabindex="-1"></a><span class="dv">5</span> <span class="fl">0.020034113</span>  <span class="fl">0.01690146</span>      <span class="dv">8</span>  <span class="fl">0.1872029</span></span>
<span id="cb6-22"><a href="#cb6-22" tabindex="-1"></a><span class="dv">6</span> <span class="fl">0.006118529</span> <span class="sc">-</span><span class="fl">0.24855631</span>    <span class="sc">-</span><span class="dv">19</span> <span class="sc">-</span><span class="fl">0.4446068</span></span></code></pre></div>
<p>What’s more, <code>biasDetect</code> function allows to return the
biased genes either in data frame format or in plots. The default format
is in data frame.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>bias <span class="ot">&lt;-</span> <span class="fu">biasDetect</span>(<span class="at">batch_df =</span> batch_df, <span class="at">nSD_dev =</span> <span class="dv">10</span>, <span class="at">nSD_rank =</span> <span class="dv">5</span>)</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>bias</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>          GADD45B             DUSP1              EGR1             NR4A1 </span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="st">"ENSG00000099860"</span> <span class="st">"ENSG00000120129"</span> <span class="st">"ENSG00000120738"</span> <span class="st">"ENSG00000123358"</span> </span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>             FOSB           GADD45G              EGR4              BTG2 </span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="st">"ENSG00000125740"</span> <span class="st">"ENSG00000130222"</span> <span class="st">"ENSG00000135625"</span> <span class="st">"ENSG00000159388"</span> </span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>              FOS              JUNB             NPAS4               JUN </span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a><span class="st">"ENSG00000170345"</span> <span class="st">"ENSG00000171223"</span> <span class="st">"ENSG00000174576"</span> <span class="st">"ENSG00000177606"</span> </span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>           HSPA1B              XIST          MTRNR2L1 </span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a><span class="st">"ENSG00000204388"</span> <span class="st">"ENSG00000229807"</span> <span class="st">"ENSG00000256618"</span> </span></code></pre></div>
<p>To see the plots, add <code>visual = TRUE</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bias_fig</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/biasDetect.html">biasDetect</a></span><span class="op">(</span>batch_df <span class="op">=</span> <span class="va">batch_df</span>, nSD_dev <span class="op">=</span> <span class="fl">10</span>, nSD_rank <span class="op">=</span> <span class="fl">5</span>, visual <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><em>Figure a) SVGs with relative change in deviance &gt;= 10 SDs from
the mean</em></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bias_fig</span><span class="op">$</span><span class="va">deviance</span></span></code></pre></div>
<p><img src="spe_files/figure-html/dev-1.png" width="700"></p>
<p><em>Figure b) SVGs with rank difference &gt;= 5 SDs from the
mean</em></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bias_fig</span><span class="op">$</span><span class="va">rank</span></span></code></pre></div>
<p><img src="spe_files/figure-html/rank-1.png" width="700"></p>
<p>Finally, we can obtain a newer <code>spe_sub4</code> object without
the biased genes.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>svgs_filt <span class="ot">=</span> <span class="fu">setdiff</span>(<span class="fu">rownames</span>(spe), bias)</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="fu">length</span>(svgs_filt)</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>[<span class="dv">1</span>] <span class="dv">2067</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="additional-information">Additional information<a class="anchor" aria-label="anchor" href="#additional-information"></a>
</h3>
<p>Please read <a href="https://github.com/jac-thom/findBiasedFeatures" class="external-link">reference
documentation</a> for detailed codes and scripts conducting PRECAST
clustering <a href="https://github.com/jac-thom/findBiasedFeatures/blob/main/code/02_spatialHPC_PRECAST-svgs.R" class="external-link">before</a>
and <a href="https://github.com/jac-thom/findBiasedFeatures/blob/main/code/05_spatialHPC_PRECAST-svgs-no-bias.R" class="external-link">after</a>
removing the biased gene.</p>
</div>
<div class="section level3 unnumbered">
<h3 class="unnumbered" id="r-session-information">
<code>R</code> session information<a class="anchor" aria-label="anchor" href="#r-session-information"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Session info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.2 (2024-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.1 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] httr_1.4.7                  SpatialExperiment_1.16.0   </span></span>
<span><span class="co">#&gt;  [3] SingleCellExperiment_1.28.1 SummarizedExperiment_1.36.0</span></span>
<span><span class="co">#&gt;  [5] Biobase_2.66.0              GenomicRanges_1.58.0       </span></span>
<span><span class="co">#&gt;  [7] GenomeInfoDb_1.42.1         IRanges_2.40.1             </span></span>
<span><span class="co">#&gt;  [9] S4Vectors_0.44.0            BiocGenerics_0.52.0        </span></span>
<span><span class="co">#&gt; [11] MatrixGenerics_1.18.1       matrixStats_1.5.0          </span></span>
<span><span class="co">#&gt; [13] BiasDetect_0.99.0           BiocStyle_2.34.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] gtable_0.3.6            rjson_0.2.23            xfun_0.50              </span></span>
<span><span class="co">#&gt;  [4] bslib_0.8.0             ggplot2_3.5.1           ggrepel_0.9.6          </span></span>
<span><span class="co">#&gt;  [7] lattice_0.22-6          vctrs_0.6.5             tools_4.4.2            </span></span>
<span><span class="co">#&gt; [10] generics_0.1.3          parallel_4.4.2          tibble_3.2.1           </span></span>
<span><span class="co">#&gt; [13] pkgconfig_2.0.3         Matrix_1.7-1            RColorBrewer_1.1-3     </span></span>
<span><span class="co">#&gt; [16] desc_1.4.3              lifecycle_1.0.4         GenomeInfoDbData_1.2.13</span></span>
<span><span class="co">#&gt; [19] farver_2.1.2            compiler_4.4.2          textshaping_1.0.0      </span></span>
<span><span class="co">#&gt; [22] munsell_0.5.1           codetools_0.2-20        htmltools_0.5.8.1      </span></span>
<span><span class="co">#&gt; [25] sass_0.4.9              yaml_2.3.10             pkgdown_2.1.1          </span></span>
<span><span class="co">#&gt; [28] pillar_1.10.1           crayon_1.5.3            jquerylib_0.1.4        </span></span>
<span><span class="co">#&gt; [31] BiocParallel_1.40.0     DelayedArray_0.32.0     cachem_1.1.0           </span></span>
<span><span class="co">#&gt; [34] magick_2.8.5            abind_1.4-8             rsvd_1.0.5             </span></span>
<span><span class="co">#&gt; [37] tidyselect_1.2.1        digest_0.6.37           BiocSingular_1.22.0    </span></span>
<span><span class="co">#&gt; [40] dplyr_1.1.4             bookdown_0.42           labeling_0.4.3         </span></span>
<span><span class="co">#&gt; [43] fastmap_1.2.0           grid_4.4.2              colorspace_2.1-1       </span></span>
<span><span class="co">#&gt; [46] cli_3.6.3               SparseArray_1.6.1       magrittr_2.0.3         </span></span>
<span><span class="co">#&gt; [49] S4Arrays_1.6.0          withr_3.0.2             UCSC.utils_1.2.0       </span></span>
<span><span class="co">#&gt; [52] scales_1.3.0            rmarkdown_2.29          XVector_0.46.0         </span></span>
<span><span class="co">#&gt; [55] ragg_1.3.3              beachmat_2.22.0         ScaledMatrix_1.14.0    </span></span>
<span><span class="co">#&gt; [58] evaluate_1.0.3          knitr_1.49              irlba_2.3.5.1          </span></span>
<span><span class="co">#&gt; [61] rlang_1.1.5             Rcpp_1.0.14             glue_1.8.0             </span></span>
<span><span class="co">#&gt; [64] BiocManager_1.30.25     scry_1.18.0             jsonlite_1.8.9         </span></span>
<span><span class="co">#&gt; [67] R6_2.5.1                systemfonts_1.2.1       fs_1.6.5               </span></span>
<span><span class="co">#&gt; [70] zlibbioc_1.52.0</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Christine Hou, Jacqui Thompson, Stephanie C. Hicks.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
