<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Find Bias Features - Spatial Transcriptomics Data â€¢ BatchSVG</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Find Bias Features - Spatial Transcriptomics Data">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">BatchSVG</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/spe.html">Find Bias Features - Spatial Transcriptomics Data</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/christinehou11/BatchSVG/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Find Bias Features - Spatial Transcriptomics Data</h1>
                        <h4 data-toc-skip class="author">Christine
Hou</h4>
            <address class="author_afil">
      Department of Biostatistics, Johns Hopkins
University<br><a class="author_email" href="mailto:#"></a><a href="mailto:chris2018hou@gmail.com" class="email">chris2018hou@gmail.com</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/christinehou11/BatchSVG/blob/main/vignettes/spe.Rmd" class="external-link"><code>vignettes/spe.Rmd</code></a></small>
      <div class="d-none name"><code>spe.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h3>
<p><code>BatchSVG</code> package was developed on an unrelated DLPFC
dataset available through the <a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-022-08601-w" class="external-link">spatialLIBD</a>
package. We first chose <strong>binomial deviance model</strong> from <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1861-6" class="external-link">scry</a>
as the feature selection method that can incorporate a batch variable
into the model. Next, we compared the per-gene ranks and dispersion
values when the model was run with and without a batch effect. Our
data-driven thresholding approach is using the cutoffs based on the
number of standard deviance (nSD) of deviance and rank difference
metrics. Based on self-selected nSD cutoffs, the genes identified as
biased can be filtered out.</p>
<p>The methodology details can be found in <a href="https://jac-thom.github.io/findBiasedFeatures/" class="external-link">Find Bias
Feature</a> written by Jacqui Thompson, and the documentation related
codes can be found on <a href="https://github.com/jac-thom/findBiasedFeatures" class="external-link">GitHub</a>.</p>
</div>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p><code>BatchSVG</code> is a R package. Install development version
from <a href="https://christinehou11.github.io/BatchSVG">GitHub</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install</span><span class="op">(</span><span class="st">"christinehou11/BatchSVG"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h3>
<p>Install <code>BatchSVG</code></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"BatchSVG"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check that you have a valid Bioconductor installation</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/valid.html" class="external-link">valid</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Load required packages</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://christinehou11.github.io/BatchSVG/">BatchSVG</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ExperimentHub</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/drighelli/SpatialExperiment" class="external-link">SpatialExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a>
</h3>
<p>The object is also saved in the <code>ExperimentHub</code> package
named <code>humanHippocampus2024</code>, and the development version
from <a href="(https://github.com/christinehou11/humanHippocampus2024)" class="external-link">GitHub</a>
is available.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ehub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html" class="external-link">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="va">ehub</span><span class="op">[[</span><span class="st">"EH9605"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">spe</span> <span class="co"># 31483, 150917 </span></span></code></pre></div>
<p>The dataset in this tutorial section is from <a href="https://github.com/LieberInstitute/spatial_hpc" class="external-link">spatialHPC</a>. We
filtered out 4 samples.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fix_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">)</span>, <span class="va">slide</span>, <span class="va">array</span>, <span class="va">brnum</span>, <span class="va">sample_id</span>, </span>
<span>    <span class="va">position</span>, <span class="va">sex</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">slide</span>, <span class="va">array</span><span class="op">)</span></span>
<span><span class="va">sub4</span> <span class="op">&lt;-</span> <span class="va">fix_order</span><span class="op">$</span><span class="va">sample_id</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">14</span>,<span class="fl">16</span>, <span class="fl">20</span>,<span class="fl">21</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="va">spe_sub4</span> <span class="op">&lt;-</span> <span class="va">spe</span><span class="op">[</span>,<span class="va">spe</span><span class="op">$</span><span class="va">sample_id</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">sub4</span><span class="op">]</span></span>
<span><span class="va">spe_sub4</span> <span class="co"># 31483, 18945</span></span></code></pre></div>
<p>We then ran <a href="https://www.nature.com/articles/s41467-023-39748-z" class="external-link">nnSVG</a> to
identify spatially-variable genes (SVGs). We filtered to only the top
2000 significant features in the 4 samples. Weâ€™re using and further
filterer to only features that are in the top 2000 of &gt;1 sample.
Finally, we obtained 2082 SVGs.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res_ranks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,<span class="st">"res_ranks.csv"</span>,package <span class="op">=</span> <span class="st">"BatchSVG"</span><span class="op">)</span>,</span>
<span>    row.names <span class="op">=</span> <span class="fl">1</span>, check.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">res_df_sub4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">res_ranks</span><span class="op">)</span>, <span class="va">var</span><span class="op">&lt;-</span><span class="st">"gene_id"</span><span class="op">)</span>, </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">res_ranks</span><span class="op">)</span>, names_to<span class="op">=</span><span class="st">"sample_id"</span>, values_to<span class="op">=</span><span class="st">"rank"</span>, </span>
<span>    values_drop_na<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">res_df2_sub4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">res_df_sub4</span>, <span class="va">sample_id</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"V11L05-333_B1"</span>,<span class="st">"V11L05-333_D1"</span>,<span class="st">"V11L05-335_D1"</span>,<span class="st">"V11L05-336_A1"</span><span class="op">)</span>,</span>
<span>        <span class="va">rank</span> <span class="op">&lt;=</span> <span class="fl">2000</span><span class="op">)</span> <span class="co"># top 2k sig features</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">res_df2_sub4</span><span class="op">)</span> <span class="co"># 7559</span></span>
<span></span>
<span><span class="va">svgs_sub4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">res_df2_sub4</span>, <span class="va">gene_id</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">tally</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">n</span><span class="op">&gt;</span><span class="fl">1</span><span class="op">)</span> <span class="co"># &gt;1 sample</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">svgs_sub4</span><span class="op">)</span> <span class="co"># 2082</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">svgs_sub4</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="bias-genes-identification">Bias Genes Identification<a class="anchor" aria-label="anchor" href="#bias-genes-identification"></a>
</h3>
<p>The data frame created from <code><a href="../reference/featureSelect.html">featureSelect()</a></code> function
includes <code>nSD_dev</code> and <code>nSD_rank</code>, and we can use
<code><a href="../reference/biasDetect.html">biasDetect()</a></code> function to identify the bias genes based on
the self-selected <code>nSD</code> integer thresholds for deviance and
rank respectively. The default threshold is 5 for deviance and 5 for
rank.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">SVGs</span> <span class="op">&lt;-</span> <span class="va">svgs_sub4</span><span class="op">$</span><span class="va">gene_id</span></span>
<span><span class="va">batch_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/featureSelect.html">featureSelect</a></span><span class="op">(</span><span class="va">spe_sub4</span>, batch_effect <span class="op">=</span> <span class="st">"sample_id"</span>, VGs <span class="op">=</span> <span class="va">SVGs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">batch_df</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">batch_df</span><span class="op">)</span></span></code></pre></div>
<p>Whatâ€™s more, <code>biasDetect</code> function allows to return the
biased genes either in data frame format or in plots. The default format
is in data frame.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bias</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/biasDetect.html">biasDetect</a></span><span class="op">(</span>batch_df <span class="op">=</span> <span class="va">batch_df</span>, nSD_dev <span class="op">=</span> <span class="fl">10</span>, nSD_rank <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="va">bias</span></span></code></pre></div>
<p>To see the plots, add <code>visual = TRUE</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bias_fig</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/biasDetect.html">biasDetect</a></span><span class="op">(</span>batch_df <span class="op">=</span> <span class="va">batch_df</span>, nSD_dev <span class="op">=</span> <span class="fl">10</span>, nSD_rank <span class="op">=</span> <span class="fl">5</span>, visual <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><em>Figure a) SVGs with relative change in deviance &gt;= 10 SDs from
the mean</em></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bias_fig</span><span class="op">$</span><span class="va">deviance</span></span></code></pre></div>
<p><em>Figure b) SVGs with rank difference &gt;= 5 SDs from the
mean</em></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bias_fig</span><span class="op">$</span><span class="va">rank</span></span></code></pre></div>
<p>Finally, we can obtain a newer <code>spe_sub4</code> object without
the biased genes.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">svgs_filt</span> <span class="op">=</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">svgs_sub4</span><span class="op">)</span>, <span class="va">bias</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">svgs_filt</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="additional-information">Additional information<a class="anchor" aria-label="anchor" href="#additional-information"></a>
</h3>
<p>Please read <a href="https://github.com/jac-thom/findBiasedFeatures" class="external-link">reference
documentation</a> for detailed codes and scripts conducting PRECAST
clustering <a href="https://github.com/jac-thom/findBiasedFeatures/blob/main/code/02_spatialHPC_PRECAST-svgs.R" class="external-link">before</a>
and <a href="https://github.com/jac-thom/findBiasedFeatures/blob/main/code/05_spatialHPC_PRECAST-svgs-no-bias.R" class="external-link">after</a>
removing the biased gene.</p>
</div>
<div class="section level3 unnumbered">
<h3 class="unnumbered" id="r-session-information">
<code>R</code> session information<a class="anchor" aria-label="anchor" href="#r-session-information"></a>
</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Session info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.2 (2024-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.1 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] tibble_3.2.1                dplyr_1.1.4                </span></span>
<span><span class="co">#&gt;  [3] tidyr_1.3.1                 SpatialExperiment_1.16.0   </span></span>
<span><span class="co">#&gt;  [5] SingleCellExperiment_1.28.1 SummarizedExperiment_1.36.0</span></span>
<span><span class="co">#&gt;  [7] Biobase_2.66.0              GenomicRanges_1.58.0       </span></span>
<span><span class="co">#&gt;  [9] GenomeInfoDb_1.42.3         IRanges_2.40.1             </span></span>
<span><span class="co">#&gt; [11] S4Vectors_0.44.0            MatrixGenerics_1.18.1      </span></span>
<span><span class="co">#&gt; [13] matrixStats_1.5.0           ExperimentHub_2.14.0       </span></span>
<span><span class="co">#&gt; [15] AnnotationHub_3.14.0        BiocFileCache_2.14.0       </span></span>
<span><span class="co">#&gt; [17] dbplyr_2.5.0                BiocGenerics_0.52.0        </span></span>
<span><span class="co">#&gt; [19] BatchSVG_0.99.1             BiocStyle_2.34.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] DBI_1.2.3               gridExtra_2.3           rlang_1.1.5            </span></span>
<span><span class="co">#&gt;  [4] magrittr_2.0.3          compiler_4.4.2          RSQLite_2.3.9          </span></span>
<span><span class="co">#&gt;  [7] png_0.1-8               systemfonts_1.2.1       vctrs_0.6.5            </span></span>
<span><span class="co">#&gt; [10] pkgconfig_2.0.3         crayon_1.5.3            fastmap_1.2.0          </span></span>
<span><span class="co">#&gt; [13] magick_2.8.5            backports_1.5.0         XVector_0.46.0         </span></span>
<span><span class="co">#&gt; [16] rmarkdown_2.29          UCSC.utils_1.2.0        ragg_1.3.3             </span></span>
<span><span class="co">#&gt; [19] purrr_1.0.4             bit_4.5.0.1             xfun_0.51              </span></span>
<span><span class="co">#&gt; [22] zlibbioc_1.52.0         cachem_1.1.0            beachmat_2.22.0        </span></span>
<span><span class="co">#&gt; [25] jsonlite_1.9.0          blob_1.2.4              DelayedArray_0.32.0    </span></span>
<span><span class="co">#&gt; [28] BiocParallel_1.40.0     broom_1.0.7             irlba_2.3.5.1          </span></span>
<span><span class="co">#&gt; [31] parallel_4.4.2          R6_2.6.1                bslib_0.9.0            </span></span>
<span><span class="co">#&gt; [34] RColorBrewer_1.1-3      car_3.1-3               jquerylib_0.1.4        </span></span>
<span><span class="co">#&gt; [37] Rcpp_1.0.14             bookdown_0.42           knitr_1.49             </span></span>
<span><span class="co">#&gt; [40] Matrix_1.7-1            tidyselect_1.2.1        abind_1.4-8            </span></span>
<span><span class="co">#&gt; [43] yaml_2.3.10             codetools_0.2-20        curl_6.2.1             </span></span>
<span><span class="co">#&gt; [46] lattice_0.22-6          scry_1.18.0             KEGGREST_1.46.0        </span></span>
<span><span class="co">#&gt; [49] evaluate_1.0.3          desc_1.4.3              Biostrings_2.74.1      </span></span>
<span><span class="co">#&gt; [52] pillar_1.10.1           BiocManager_1.30.25     ggpubr_0.6.0           </span></span>
<span><span class="co">#&gt; [55] filelock_1.0.3          carData_3.0-5           generics_0.1.3         </span></span>
<span><span class="co">#&gt; [58] BiocVersion_3.20.0      ggplot2_3.5.1           munsell_0.5.1          </span></span>
<span><span class="co">#&gt; [61] scales_1.3.0            glue_1.8.0              tools_4.4.2            </span></span>
<span><span class="co">#&gt; [64] ScaledMatrix_1.14.0     ggsignif_0.6.4          fs_1.6.5               </span></span>
<span><span class="co">#&gt; [67] grid_4.4.2              AnnotationDbi_1.68.0    colorspace_2.1-1       </span></span>
<span><span class="co">#&gt; [70] GenomeInfoDbData_1.2.13 BiocSingular_1.22.0     Formula_1.2-5          </span></span>
<span><span class="co">#&gt; [73] cli_3.6.4               rsvd_1.0.5              rappdirs_0.3.3         </span></span>
<span><span class="co">#&gt; [76] textshaping_1.0.0       S4Arrays_1.6.0          gtable_0.3.6           </span></span>
<span><span class="co">#&gt; [79] rstatix_0.7.2           sass_0.4.9              digest_0.6.37          </span></span>
<span><span class="co">#&gt; [82] SparseArray_1.6.1       ggrepel_0.9.6           rjson_0.2.23           </span></span>
<span><span class="co">#&gt; [85] memoise_2.0.1           htmltools_0.5.8.1       pkgdown_2.1.1          </span></span>
<span><span class="co">#&gt; [88] lifecycle_1.0.4         httr_1.4.7              bit64_4.6.0-1</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Christine Hou, Jacqui Thompson, Stephanie C. Hicks.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
