<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Find Batch-biased Features in SVGs • BatchSVG</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Find Batch-biased Features in SVGs">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">BatchSVG</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.8</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/spe.html">Find Batch-biased Features in SVGs</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/christinehou11/BatchSVG/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Find Batch-biased Features in SVGs</h1>
                        <h4 data-toc-skip class="author">Christine
Hou</h4>
            <address class="author_afil">
      Department of Biostatistics, Johns Hopkins
University<br><a class="author_email" href="mailto:#"></a><a href="mailto:chris2018hou@gmail.com" class="email">chris2018hou@gmail.com</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/christinehou11/BatchSVG/blob/main/vignettes/spe.Rmd" class="external-link"><code>vignettes/spe.Rmd</code></a></small>
      <div class="d-none name"><code>spe.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h3>
<p><code>BatchSVG</code> is the R/Bioconductor package for spatial
transcriptomics data quality control (QC). As the feature-based QC
method, the package provides functions to identify the biased features
associated with the batch effect(s) (e.g. sample, slide, and sex) in
spatially variable genes (SVGs) using binomial deviance model, aiming to
develop the downstream clustering performances and remove the technical
noises caused by batch effects. The package works with <a href="https://github.com/drighelli/SpatialExperiment" class="external-link">SpatialExperiment</a>
objects.</p>
</div>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>(After accepted in <a href="https://bioconductor.org/" class="external-link">Bioconductor</a>).</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"BatchSVG"</span><span class="op">)</span></span></code></pre></div>
<p>Install the development version from <a href="https://christinehou11.github.io/BatchSVG">GitHub</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install</span><span class="op">(</span><span class="st">"christinehou11/BatchSVG"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="biased-feature-identification">Biased Feature Identification<a class="anchor" aria-label="anchor" href="#biased-feature-identification"></a>
</h3>
<p>In this section, we will include the standard workflow for using
<code>BatchSVG</code> to show how the method help to detect and
visualize the biased features in SVGs.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/christinehou11/BatchSVG" class="external-link">BatchSVG</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/LieberInstitute/spatialLIBD" class="external-link">spatialLIBD</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a>
</h4>
<p>We will use the`spatially-resolved transcriptomics (SRT) dataset from
the <a href="https://research.libd.org/spatialLIBD/" class="external-link">spatialLIBD</a>
package.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>spatialLIBD_spe <span class="ot">&lt;-</span> <span class="fu">fetch_data</span>(<span class="at">type =</span> <span class="st">"spe"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>spatialLIBD_spe</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>class<span class="sc">:</span> SpatialExperiment </span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>dim<span class="sc">:</span> <span class="dv">33538</span> <span class="dv">47681</span> </span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="fu">metadata</span>(<span class="dv">0</span>)<span class="sc">:</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="fu">assays</span>(<span class="dv">2</span>)<span class="sc">:</span> counts logcounts</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="fu">rownames</span>(<span class="dv">33538</span>)<span class="sc">:</span> ENSG00000243485 ENSG00000237613 ... ENSG00000277475</span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>  ENSG00000268674</span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a>rowData <span class="fu">names</span>(<span class="dv">9</span>)<span class="sc">:</span> source type ... gene_search is_top_hvg</span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a><span class="fu">colnames</span>(<span class="dv">47681</span>)<span class="sc">:</span> AAACAACGAATAGTTC<span class="dv">-1</span> AAACAAGTATCTCCCA<span class="dv">-1</span> ...</span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>  TTGTTTCCATACAACT<span class="dv">-1</span> TTGTTTGTGTAAATTC<span class="dv">-1</span></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a>colData <span class="fu">names</span>(<span class="dv">69</span>)<span class="sc">:</span> sample_id Cluster ... array_row array_col</span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a><span class="fu">reducedDimNames</span>(<span class="dv">6</span>)<span class="sc">:</span> PCA TSNE_perplexity50 ... TSNE_perplexity80</span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a>  UMAP_neighbors15</span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a>mainExpName<span class="sc">:</span> <span class="cn">NULL</span></span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a><span class="fu">altExpNames</span>(<span class="dv">0</span>)<span class="sc">:</span></span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a>spatialCoords <span class="fu">names</span>(<span class="dv">2</span>) <span class="sc">:</span> pxl_col_in_fullres pxl_row_in_fullres</span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a>imgData <span class="fu">names</span>(<span class="dv">4</span>)<span class="sc">:</span> sample_id image_id data scaleFactor</span></code></pre></div>
<p>We will use the spatially variable genes set generated. The result is
generated from <a href="(https://www.nature.com/articles/s41467-023-39748-z)" class="external-link">nnSVG</a>
package.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">libd_nnsvgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,<span class="st">"libd-all_nnSVG_p-05-features-df.csv"</span>,</span>
<span>        package <span class="op">=</span> <span class="st">"BatchSVG"</span><span class="op">)</span>,</span>
<span>    row.names <span class="op">=</span> <span class="fl">1</span>, check.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="perform-feature-selection-using-featureselect">Perform Feature Selection using <code>featureSelect()</code><a class="anchor" aria-label="anchor" href="#perform-feature-selection-using-featureselect"></a>
</h4>
<p>We will perform feature selection on a subset of spatial
transcriptomics data (<em>input</em>) using a predefined set of
spatially variable genes (<em>VGs</em>). Specifically, we will compute
the number of standard deviations for the relative change in deviance
(<strong>nSD_dev_{batch effect}</strong>) and rank difference
(<strong>nSD_rank_{batch effect}</strong>) before and after adjusting
for batch effects.</p>
<p>The <code><a href="../reference/featureSelect.html">featureSelect()</a></code> function enables feature selection
while accounting for multiple batch effects. It returns a
<strong>list</strong> of data frames, where each batch effect is
associated with a corresponding data frame containing key results,
including:</p>
<ul>
<li><p>Relative change in deviance before and after batch effect
adjustment</p></li>
<li><p>Rank differences between the batch-corrected and uncorrected
results</p></li>
<li><p>Number of standard deviations (nSD) for both relative change in
deviance and rank difference</p></li>
</ul>
<p>We will use the example of applying <code><a href="../reference/featureSelect.html">featureSelect()</a></code> to
the sample dataset while adjusting for the batch effect of
<em>subject</em>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>list_batch_df <span class="ot">&lt;-</span> <span class="fu">featureSelect</span>(<span class="at">input =</span> spatialLIBD_spe, </span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>    <span class="at">batch_effect =</span> <span class="st">"subject"</span>, <span class="at">VGs =</span> libd_nnsvgs<span class="sc">$</span>gene_id)</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>Running feature selection without batch...</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>Batch Effect<span class="sc">:</span> subject</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>Running feature selection without batch...</span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>Calculating deviance and rank difference...</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">class</span>(list_batch_df)</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">"list"</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">head</span>(list_batch_df<span class="sc">$</span>subject)</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>          gene_id gene_name dev_default rank_default dev_subject rank_subject</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="dv">1</span> ENSG00000187608     ISG15    <span class="fl">43591.43</span>         <span class="dv">1151</span>    <span class="fl">43304.39</span>         <span class="dv">1157</span></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="dv">2</span> ENSG00000131584     ACAP3    <span class="fl">39115.97</span>         <span class="dv">1498</span>    <span class="fl">38826.05</span>         <span class="dv">1509</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a><span class="dv">3</span> ENSG00000242485    MRPL20    <span class="fl">47074.23</span>          <span class="dv">920</span>    <span class="fl">46912.53</span>          <span class="dv">918</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="dv">4</span> ENSG00000160075     SSU72    <span class="fl">47815.13</span>          <span class="dv">862</span>    <span class="fl">47742.10</span>          <span class="dv">852</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="dv">5</span> ENSG00000078369      GNB1    <span class="fl">54464.75</span>          <span class="dv">477</span>    <span class="fl">54385.81</span>          <span class="dv">462</span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="dv">6</span> ENSG00000187730     GABRD    <span class="fl">51521.13</span>          <span class="dv">646</span>    <span class="fl">51099.35</span>          <span class="dv">648</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>       d_diff nSD_dev_subject r_diff nSD_rank_subject</span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a><span class="dv">1</span> <span class="fl">0.006628419</span>     <span class="sc">-</span><span class="fl">0.04505973</span>      <span class="dv">6</span>       <span class="fl">0.15937488</span></span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a><span class="dv">2</span> <span class="fl">0.007467317</span>     <span class="sc">-</span><span class="fl">0.03697398</span>     <span class="dv">11</span>       <span class="fl">0.29218728</span></span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a><span class="dv">3</span> <span class="fl">0.003446929</span>     <span class="sc">-</span><span class="fl">0.07572463</span>     <span class="sc">-</span><span class="dv">2</span>      <span class="sc">-</span><span class="fl">0.05312496</span></span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a><span class="dv">4</span> <span class="fl">0.001529688</span>     <span class="sc">-</span><span class="fl">0.09420402</span>    <span class="sc">-</span><span class="dv">10</span>      <span class="sc">-</span><span class="fl">0.26562480</span></span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a><span class="dv">5</span> <span class="fl">0.001451448</span>     <span class="sc">-</span><span class="fl">0.09495814</span>    <span class="sc">-</span><span class="dv">15</span>      <span class="sc">-</span><span class="fl">0.39843720</span></span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a><span class="dv">6</span> <span class="fl">0.008254157</span>     <span class="sc">-</span><span class="fl">0.02939000</span>      <span class="dv">2</span>       <span class="fl">0.05312496</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="visualize-svg-selection-using-svg_nsd-for-batch-effects">Visualize SVG Selection Using <code>svg_nSD</code> for Batch
Effects<a class="anchor" aria-label="anchor" href="#visualize-svg-selection-using-svg_nsd-for-batch-effects"></a>
</h4>
<p>The <code><a href="../reference/svg_nSD.html">svg_nSD()</a></code> function generates visualizations to
assess batch effects in spatially variable genes (SVGs). It produces bar
charts showing the distribution of SVGs based on relative change in
deviance and rank difference, with colors representing different nSD
intervals. Additionally, scatter plots compare deviance and rank values
with and without batch effects.</p>
<p>By interpreting these plots, we can determine appropriate nSD
thresholds for filtering biased features. The left panels illustrate the
distribution of SVGs in terms of deviance and rank difference, while the
right panels compare values before and after accounting for batch
effects.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/svg_nSD.html">svg_nSD</a></span><span class="op">(</span>list_batch_df <span class="op">=</span> <span class="va">list_batch_df</span>, </span>
<span>            sd_interval_dev <span class="op">=</span> <span class="fl">3</span>, sd_interval_rank <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p><em>Figure 1. Visualizations of nSD_dev and nSD_rank threshold
selection</em></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plots</span><span class="op">$</span><span class="va">subject</span></span></code></pre></div>
<p><img src="spe_files/figure-html/figure%201-1.png" width="960"></p>
</div>
<div class="section level4">
<h4 id="identify-biased-genes-using-biasdetect">Identify Biased Genes Using <code>biasDetect()</code><a class="anchor" aria-label="anchor" href="#identify-biased-genes-using-biasdetect"></a>
</h4>
<p>The function <code><a href="../reference/biasDetect.html">biasDetect()</a></code> is designed to identify and
filter out biased genes across different batch effects. Using threshold
values selected from the visualization results generated by
<code><a href="../reference/svg_nSD.html">svg_nSD()</a></code>, this function systematically detects outliers
that exceed a specified number of standard deviation (nSD) threshold in
either relative deviance change, rank difference, or both.</p>
<p>The function outputs visualizations comparing deviance and rank
values with and without batch effects. Genes with high deviations,
highlighted in color, are identified as potentially biased and can be
excluded based on the selected nSD thresholds.</p>
<p>The function offers flexibility in customizing the plot aesthetics,
allowing users to adjust the data point size
(<strong>plot_point_size</strong>), shape
(<strong>plot_point_shape</strong>), annotated text size
(<strong>plot_text_size</strong>), and data point color palette
(<strong>plot_palette</strong>). Default values are provided for these
parameters if not specified. Users should refer to <a href="https://ggplot2.tidyverse.org/index.html" class="external-link">ggplot2</a> aesthetic
guidelines to ensure appropriate values are assigned for each
parameter.</p>
<p>We will use <code>nSD_dev = 3</code> and <code>nSD_rank = 3</code> as
the example. The user should adjust the value based on their dataset
features.</p>
<p><strong>Usage of Different Threshold Options</strong></p>
<ul>
<li>
<code>threshold = "dev"</code>: Filters biased genes based only on
the relative change in deviance. Genes with deviance changes exceeding
the specified <code>nSD_dev</code> threshold are identified as
batch-affected and can be removed.</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bias_dev</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/biasDetect.html">biasDetect</a></span><span class="op">(</span>list_batch_df <span class="op">=</span> <span class="va">list_batch_df</span>, </span>
<span>    threshold <span class="op">=</span> <span class="st">"dev"</span>, nSD_dev <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p><em>Table 1. Outlier Genes defined by nSD_dev only</em></p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="fu">head</span>(bias_dev<span class="sc">$</span>subject<span class="sc">$</span>Table)</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>          gene_id gene_name dev_default rank_default dev_subject rank_subject</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a><span class="dv">1</span> ENSG00000255823  MTRNR2L8    <span class="fl">286927.9</span>            <span class="dv">5</span>    <span class="fl">67738.95</span>           <span class="dv">77</span></span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a><span class="dv">2</span> ENSG00000087250       MT3    <span class="fl">115108.2</span>           <span class="dv">21</span>    <span class="fl">86093.50</span>           <span class="dv">31</span></span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a><span class="dv">3</span> ENSG00000256618  MTRNR2L1    <span class="fl">167381.9</span>           <span class="dv">14</span>    <span class="fl">40299.65</span>         <span class="dv">1368</span></span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a><span class="dv">4</span> ENSG00000198840    MT<span class="sc">-</span>ND3    <span class="fl">170192.2</span>           <span class="dv">12</span>   <span class="fl">120647.03</span>           <span class="dv">17</span></span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a>     d_diff nSD_dev_subject r_diff nSD_rank_subject nSD_bin_dev dev_outlier</span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a><span class="dv">1</span> <span class="fl">3.2357886</span>       <span class="fl">31.079310</span>     <span class="dv">72</span>        <span class="fl">1.9124986</span>     [<span class="dv">30</span>,<span class="dv">33</span>]        <span class="cn">TRUE</span></span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a><span class="dv">2</span> <span class="fl">0.3370142</span>        <span class="fl">3.139375</span>     <span class="dv">10</span>        <span class="fl">0.2656248</span>       [<span class="dv">3</span>,<span class="dv">6</span><span class="er">)</span>        <span class="cn">TRUE</span></span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a><span class="dv">3</span> <span class="fl">3.1534316</span>       <span class="fl">30.285509</span>   <span class="dv">1354</span>       <span class="fl">35.9655982</span>     [<span class="dv">30</span>,<span class="dv">33</span>]        <span class="cn">TRUE</span></span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a><span class="dv">4</span> <span class="fl">0.4106625</span>        <span class="fl">3.849237</span>      <span class="dv">5</span>        <span class="fl">0.1328124</span>       [<span class="dv">3</span>,<span class="dv">6</span><span class="er">)</span>        <span class="cn">TRUE</span></span></code></pre></div>
<p>We can change the data point size using
<strong>plot_point_size</strong>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># size default = 3</span></span>
<span><span class="va">bias_dev_size</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/biasDetect.html">biasDetect</a></span><span class="op">(</span>list_batch_df <span class="op">=</span> <span class="va">list_batch_df</span>, </span>
<span>    threshold <span class="op">=</span> <span class="st">"dev"</span>, nSD_dev <span class="op">=</span> <span class="fl">3</span>, plot_point_size <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<p><em>Figure 2. Customize point size</em></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span><span class="va">bias_dev</span><span class="op">$</span><span class="va">subject</span><span class="op">$</span><span class="va">Plot</span>, <span class="va">bias_dev_size</span><span class="op">$</span><span class="va">subject</span><span class="op">$</span><span class="va">Plot</span><span class="op">)</span></span></code></pre></div>
<p><img src="spe_files/figure-html/figure%202-1.png" width="960"></p>
<ul>
<li>
<code>threshold = "rank"</code>: Identifies biased genes based
solely on rank difference. Genes with rank shifts exceeding
<code>nSD_rank</code> are considered biased.</li>
</ul>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bias_rank</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/biasDetect.html">biasDetect</a></span><span class="op">(</span>list_batch_df <span class="op">=</span> <span class="va">list_batch_df</span>, </span>
<span>    threshold <span class="op">=</span> <span class="st">"rank"</span>, nSD_rank <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p><em>Table 2. Outlier Genes defined by nSD_rank only</em></p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="fu">head</span>(bias_rank<span class="sc">$</span>subject<span class="sc">$</span>Table)</span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>          gene_id gene_name dev_default rank_default dev_subject rank_subject</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a><span class="dv">1</span> ENSG00000184144     CNTN2    <span class="fl">44388.43</span>         <span class="dv">1101</span>    <span class="fl">42016.77</span>         <span class="dv">1258</span></span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a><span class="dv">2</span> ENSG00000136541      ERMN    <span class="fl">48106.50</span>          <span class="dv">844</span>    <span class="fl">44879.30</span>         <span class="dv">1036</span></span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a><span class="dv">3</span> ENSG00000013297    CLDN11    <span class="fl">53385.84</span>          <span class="dv">539</span>    <span class="fl">49776.21</span>          <span class="dv">732</span></span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a><span class="dv">4</span> ENSG00000164124   TMEM144    <span class="fl">39407.07</span>         <span class="dv">1471</span>    <span class="fl">37587.88</span>         <span class="dv">1605</span></span>
<span id="cb16-7"><a href="#cb16-7" tabindex="-1"></a><span class="dv">5</span> ENSG00000204655       MOG    <span class="fl">37293.40</span>         <span class="dv">1648</span>    <span class="fl">35613.38</span>         <span class="dv">1762</span></span>
<span id="cb16-8"><a href="#cb16-8" tabindex="-1"></a><span class="dv">6</span> ENSG00000091129     NRCAM    <span class="fl">51388.42</span>          <span class="dv">655</span>    <span class="fl">49006.18</span>          <span class="dv">776</span></span>
<span id="cb16-9"><a href="#cb16-9" tabindex="-1"></a>      d_diff nSD_dev_subject r_diff nSD_rank_subject nSD_bin_rank rank_outlier</span>
<span id="cb16-10"><a href="#cb16-10" tabindex="-1"></a><span class="dv">1</span> <span class="fl">0.05644564</span>       <span class="fl">0.4351052</span>    <span class="dv">157</span>         <span class="fl">4.170309</span>        [<span class="dv">3</span>,<span class="dv">6</span><span class="er">)</span>         <span class="cn">TRUE</span></span>
<span id="cb16-11"><a href="#cb16-11" tabindex="-1"></a><span class="dv">2</span> <span class="fl">0.07190853</span>       <span class="fl">0.5841448</span>    <span class="dv">192</span>         <span class="fl">5.099996</span>        [<span class="dv">3</span>,<span class="dv">6</span><span class="er">)</span>         <span class="cn">TRUE</span></span>
<span id="cb16-12"><a href="#cb16-12" tabindex="-1"></a><span class="dv">3</span> <span class="fl">0.07251725</span>       <span class="fl">0.5900120</span>    <span class="dv">193</span>         <span class="fl">5.126559</span>        [<span class="dv">3</span>,<span class="dv">6</span><span class="er">)</span>         <span class="cn">TRUE</span></span>
<span id="cb16-13"><a href="#cb16-13" tabindex="-1"></a><span class="dv">4</span> <span class="fl">0.04839817</span>       <span class="fl">0.3575395</span>    <span class="dv">134</span>         <span class="fl">3.559372</span>        [<span class="dv">3</span>,<span class="dv">6</span><span class="er">)</span>         <span class="cn">TRUE</span></span>
<span id="cb16-14"><a href="#cb16-14" tabindex="-1"></a><span class="dv">5</span> <span class="fl">0.04717397</span>       <span class="fl">0.3457399</span>    <span class="dv">114</span>         <span class="fl">3.028123</span>        [<span class="dv">3</span>,<span class="dv">6</span><span class="er">)</span>         <span class="cn">TRUE</span></span>
<span id="cb16-15"><a href="#cb16-15" tabindex="-1"></a><span class="dv">6</span> <span class="fl">0.04861096</span>       <span class="fl">0.3595904</span>    <span class="dv">121</span>         <span class="fl">3.214060</span>        [<span class="dv">3</span>,<span class="dv">6</span><span class="er">)</span>         <span class="cn">TRUE</span></span></code></pre></div>
<p>We can change the data point shape using
<strong>plot_point_shape</strong>.</p>
<p><em>Figure 3. Customize point shape</em></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># shape default = 16</span></span>
<span><span class="va">bias_rank_shape</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/biasDetect.html">biasDetect</a></span><span class="op">(</span>list_batch_df <span class="op">=</span> <span class="va">list_batch_df</span>, </span>
<span>    threshold <span class="op">=</span> <span class="st">"rank"</span>, nSD_rank <span class="op">=</span> <span class="fl">3</span>, plot_point_shape <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span><span class="va">bias_rank</span><span class="op">$</span><span class="va">subject</span><span class="op">$</span><span class="va">Plot</span>, <span class="va">bias_rank_shape</span><span class="op">$</span><span class="va">subject</span><span class="op">$</span><span class="va">Plot</span><span class="op">)</span></span></code></pre></div>
<p><img src="spe_files/figure-html/figure%203-1.png" width="960"></p>
<ul>
<li>
<code>threshold = "both"</code>: Detects biased genes based on both
deviance change and rank difference, providing a more stringent
filtering approach.</li>
</ul>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bias_both</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/biasDetect.html">biasDetect</a></span><span class="op">(</span>list_batch_df <span class="op">=</span> <span class="va">list_batch_df</span>, threshold <span class="op">=</span> <span class="st">"both"</span>,</span>
<span>    nSD_dev <span class="op">=</span> <span class="fl">3</span>, nSD_rank <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p><em>Table 3. Outlier Genes defined by nSD_dev and nSD_rank</em></p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="fu">head</span>(bias_both<span class="sc">$</span>subject<span class="sc">$</span>Table)</span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a>          gene_id gene_name dev_default rank_default dev_subject rank_subject</span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a><span class="dv">1</span> ENSG00000184144     CNTN2    <span class="fl">44388.43</span>         <span class="dv">1101</span>    <span class="fl">42016.77</span>         <span class="dv">1258</span></span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a><span class="dv">2</span> ENSG00000136541      ERMN    <span class="fl">48106.50</span>          <span class="dv">844</span>    <span class="fl">44879.30</span>         <span class="dv">1036</span></span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a><span class="dv">3</span> ENSG00000013297    CLDN11    <span class="fl">53385.84</span>          <span class="dv">539</span>    <span class="fl">49776.21</span>          <span class="dv">732</span></span>
<span id="cb19-6"><a href="#cb19-6" tabindex="-1"></a><span class="dv">4</span> ENSG00000164124   TMEM144    <span class="fl">39407.07</span>         <span class="dv">1471</span>    <span class="fl">37587.88</span>         <span class="dv">1605</span></span>
<span id="cb19-7"><a href="#cb19-7" tabindex="-1"></a><span class="dv">5</span> ENSG00000204655       MOG    <span class="fl">37293.40</span>         <span class="dv">1648</span>    <span class="fl">35613.38</span>         <span class="dv">1762</span></span>
<span id="cb19-8"><a href="#cb19-8" tabindex="-1"></a><span class="dv">6</span> ENSG00000091129     NRCAM    <span class="fl">51388.42</span>          <span class="dv">655</span>    <span class="fl">49006.18</span>          <span class="dv">776</span></span>
<span id="cb19-9"><a href="#cb19-9" tabindex="-1"></a>      d_diff nSD_dev_subject r_diff nSD_rank_subject nSD_bin_dev dev_outlier</span>
<span id="cb19-10"><a href="#cb19-10" tabindex="-1"></a><span class="dv">1</span> <span class="fl">0.05644564</span>       <span class="fl">0.4351052</span>    <span class="dv">157</span>         <span class="fl">4.170309</span>       [<span class="dv">0</span>,<span class="dv">3</span><span class="er">)</span>       <span class="cn">FALSE</span></span>
<span id="cb19-11"><a href="#cb19-11" tabindex="-1"></a><span class="dv">2</span> <span class="fl">0.07190853</span>       <span class="fl">0.5841448</span>    <span class="dv">192</span>         <span class="fl">5.099996</span>       [<span class="dv">0</span>,<span class="dv">3</span><span class="er">)</span>       <span class="cn">FALSE</span></span>
<span id="cb19-12"><a href="#cb19-12" tabindex="-1"></a><span class="dv">3</span> <span class="fl">0.07251725</span>       <span class="fl">0.5900120</span>    <span class="dv">193</span>         <span class="fl">5.126559</span>       [<span class="dv">0</span>,<span class="dv">3</span><span class="er">)</span>       <span class="cn">FALSE</span></span>
<span id="cb19-13"><a href="#cb19-13" tabindex="-1"></a><span class="dv">4</span> <span class="fl">0.04839817</span>       <span class="fl">0.3575395</span>    <span class="dv">134</span>         <span class="fl">3.559372</span>       [<span class="dv">0</span>,<span class="dv">3</span><span class="er">)</span>       <span class="cn">FALSE</span></span>
<span id="cb19-14"><a href="#cb19-14" tabindex="-1"></a><span class="dv">5</span> <span class="fl">0.04717397</span>       <span class="fl">0.3457399</span>    <span class="dv">114</span>         <span class="fl">3.028123</span>       [<span class="dv">0</span>,<span class="dv">3</span><span class="er">)</span>       <span class="cn">FALSE</span></span>
<span id="cb19-15"><a href="#cb19-15" tabindex="-1"></a><span class="dv">6</span> <span class="fl">0.04861096</span>       <span class="fl">0.3595904</span>    <span class="dv">121</span>         <span class="fl">3.214060</span>       [<span class="dv">0</span>,<span class="dv">3</span><span class="er">)</span>       <span class="cn">FALSE</span></span>
<span id="cb19-16"><a href="#cb19-16" tabindex="-1"></a>  nSD_bin_rank rank_outlier</span>
<span id="cb19-17"><a href="#cb19-17" tabindex="-1"></a><span class="dv">1</span>        [<span class="dv">3</span>,<span class="dv">6</span><span class="er">)</span>         <span class="cn">TRUE</span></span>
<span id="cb19-18"><a href="#cb19-18" tabindex="-1"></a><span class="dv">2</span>        [<span class="dv">3</span>,<span class="dv">6</span><span class="er">)</span>         <span class="cn">TRUE</span></span>
<span id="cb19-19"><a href="#cb19-19" tabindex="-1"></a><span class="dv">3</span>        [<span class="dv">3</span>,<span class="dv">6</span><span class="er">)</span>         <span class="cn">TRUE</span></span>
<span id="cb19-20"><a href="#cb19-20" tabindex="-1"></a><span class="dv">4</span>        [<span class="dv">3</span>,<span class="dv">6</span><span class="er">)</span>         <span class="cn">TRUE</span></span>
<span id="cb19-21"><a href="#cb19-21" tabindex="-1"></a><span class="dv">5</span>        [<span class="dv">3</span>,<span class="dv">6</span><span class="er">)</span>         <span class="cn">TRUE</span></span>
<span id="cb19-22"><a href="#cb19-22" tabindex="-1"></a><span class="dv">6</span>        [<span class="dv">3</span>,<span class="dv">6</span><span class="er">)</span>         <span class="cn">TRUE</span></span></code></pre></div>
<p>We can change the data point color using
<strong>plot_palette</strong>. The color palette <a href="https://r-graph-gallery.com/38-rcolorbrewers-palettes.html" class="external-link">here</a>
can be referenced on since the function uses <code>RColorBrewer</code>
to generate colors.</p>
<p><em>Figure 4. Customize the palette color</em></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># color default = "YlOrRd"</span></span>
<span><span class="va">bias_both_color</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/biasDetect.html">biasDetect</a></span><span class="op">(</span>list_batch_df <span class="op">=</span> <span class="va">list_batch_df</span>, </span>
<span>    threshold <span class="op">=</span> <span class="st">"both"</span>, nSD_dev <span class="op">=</span> <span class="fl">3</span>, nSD_rank <span class="op">=</span> <span class="fl">3</span>, plot_palette <span class="op">=</span> <span class="st">"Greens"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span><span class="va">bias_both</span><span class="op">$</span><span class="va">subject</span><span class="op">$</span><span class="va">Plot</span>, <span class="va">bias_both_color</span><span class="op">$</span><span class="va">subject</span><span class="op">$</span><span class="va">Plot</span>,nrow <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="spe_files/figure-html/figure%204-1.png" width="960"></p>
<p>We can change the text size using <strong>plot_text_size</strong>. We
also specify the color palettes for both batch effects at the same
time.</p>
<p><em>Figure 5. Customize text size and color palette</em></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># text size default = 3</span></span>
<span><span class="va">bias_both_color_text</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/biasDetect.html">biasDetect</a></span><span class="op">(</span>list_batch_df <span class="op">=</span> <span class="va">list_batch_df</span>, </span>
<span>    threshold <span class="op">=</span> <span class="st">"both"</span>, nSD_dev <span class="op">=</span> <span class="fl">3</span>, nSD_rank <span class="op">=</span> <span class="fl">3</span>, </span>
<span>    plot_palette <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Blues"</span><span class="op">)</span>, plot_text_size <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span><span class="va">bias_both</span><span class="op">$</span><span class="va">subject</span><span class="op">$</span><span class="va">Plot</span>, <span class="va">bias_both_color_text</span><span class="op">$</span><span class="va">subject</span><span class="op">$</span><span class="va">Plot</span>,nrow <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="spe_files/figure-html/figure%205-1.png" width="960"></p>
</div>
<div class="section level4">
<h4 id="refine-svgs-by-removing-batch-affected-outliers">Refine SVGs by Removing Batch-Affected Outliers<a class="anchor" aria-label="anchor" href="#refine-svgs-by-removing-batch-affected-outliers"></a>
</h4>
<p>Finally, we obtain a refined set of spatially variable genes (SVGs)
by removing the identified outliers based on user-defined thresholds for
<code>nSD_dev</code> and <code>nSD_rank</code>.</p>
<p>Here, we use the results from bias_both, which applied
<code>threshold = "both"</code> to account for both deviance and rank
differences, with the batch effect set to sample ID.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a>bias_both_df <span class="ot">&lt;-</span> bias_both<span class="sc">$</span>subject<span class="sc">$</span>Table</span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>svgs_filt <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(libd_nnsvgs<span class="sc">$</span>gene_id, bias_both_df<span class="sc">$</span>gene_id)</span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a>svgs_filt_spe <span class="ot">&lt;-</span> libd_nnsvgs[libd_nnsvgs<span class="sc">$</span>gene_id <span class="sc">%in%</span> svgs_filt, ]</span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a><span class="fu">nrow</span>(svgs_filt_spe)</span>
<span id="cb22-5"><a href="#cb22-5" tabindex="-1"></a>[<span class="dv">1</span>] <span class="dv">1951</span></span></code></pre></div>
<p>After obtaining the refined set of SVGs, these genes can be further
analyzed using established spatial transcriptomics clustering algorithms
to explore tissue layers and spatial organization.</p>
</div>
</div>
<div class="section level3 unnumbered">
<h3 class="unnumbered" id="r-session-information">
<code>R</code> session information<a class="anchor" aria-label="anchor" href="#r-session-information"></a>
</h3>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Session info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.3 (2025-02-28)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] cowplot_1.1.3               spatialLIBD_1.18.0         </span></span>
<span><span class="co">#&gt;  [3] SpatialExperiment_1.16.0    SingleCellExperiment_1.28.1</span></span>
<span><span class="co">#&gt;  [5] SummarizedExperiment_1.36.0 Biobase_2.66.0             </span></span>
<span><span class="co">#&gt;  [7] GenomicRanges_1.58.0        GenomeInfoDb_1.42.3        </span></span>
<span><span class="co">#&gt;  [9] IRanges_2.40.1              S4Vectors_0.44.0           </span></span>
<span><span class="co">#&gt; [11] BiocGenerics_0.52.0         MatrixGenerics_1.18.1      </span></span>
<span><span class="co">#&gt; [13] matrixStats_1.5.0           BatchSVG_0.99.8            </span></span>
<span><span class="co">#&gt; [15] BiocStyle_2.34.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3       jsonlite_2.0.0           magrittr_2.0.3          </span></span>
<span><span class="co">#&gt;   [4] ggbeeswarm_0.7.2         magick_2.8.6             farver_2.1.2            </span></span>
<span><span class="co">#&gt;   [7] rmarkdown_2.29           BiocIO_1.16.0            fs_1.6.5                </span></span>
<span><span class="co">#&gt;  [10] fields_16.3.1            zlibbioc_1.52.0          ragg_1.3.3              </span></span>
<span><span class="co">#&gt;  [13] vctrs_0.6.5              Rsamtools_2.22.0         config_0.3.2            </span></span>
<span><span class="co">#&gt;  [16] memoise_2.0.1            RCurl_1.98-1.17          paletteer_1.6.0         </span></span>
<span><span class="co">#&gt;  [19] benchmarkme_1.0.8        htmltools_0.5.8.1        S4Arrays_1.6.0          </span></span>
<span><span class="co">#&gt;  [22] AnnotationHub_3.14.0     curl_6.2.2               BiocNeighbors_2.0.1     </span></span>
<span><span class="co">#&gt;  [25] SparseArray_1.6.2        sass_0.4.9               bslib_0.9.0             </span></span>
<span><span class="co">#&gt;  [28] htmlwidgets_1.6.4        desc_1.4.3               plotly_4.10.4           </span></span>
<span><span class="co">#&gt;  [31] cachem_1.1.0             GenomicAlignments_1.42.0 mime_0.13               </span></span>
<span><span class="co">#&gt;  [34] lifecycle_1.0.4          iterators_1.0.14         pkgconfig_2.0.3         </span></span>
<span><span class="co">#&gt;  [37] rsvd_1.0.5               Matrix_1.7-2             R6_2.6.1                </span></span>
<span><span class="co">#&gt;  [40] fastmap_1.2.0            GenomeInfoDbData_1.2.13  shiny_1.10.0            </span></span>
<span><span class="co">#&gt;  [43] digest_0.6.37            colorspace_2.1-1         rematch2_2.1.2          </span></span>
<span><span class="co">#&gt;  [46] AnnotationDbi_1.68.0     scater_1.34.1            irlba_2.3.5.1           </span></span>
<span><span class="co">#&gt;  [49] ExperimentHub_2.14.0     textshaping_1.0.0        RSQLite_2.3.9           </span></span>
<span><span class="co">#&gt;  [52] beachmat_2.22.0          labeling_0.4.3           filelock_1.0.3          </span></span>
<span><span class="co">#&gt;  [55] httr_1.4.7               abind_1.4-8              compiler_4.4.3          </span></span>
<span><span class="co">#&gt;  [58] withr_3.0.2              attempt_0.3.1            bit64_4.6.0-1           </span></span>
<span><span class="co">#&gt;  [61] doParallel_1.0.17        BiocParallel_1.40.0      viridis_0.6.5           </span></span>
<span><span class="co">#&gt;  [64] DBI_1.2.3                maps_3.4.2.1             sessioninfo_1.2.3       </span></span>
<span><span class="co">#&gt;  [67] rappdirs_0.3.3           DelayedArray_0.32.0      rjson_0.2.23            </span></span>
<span><span class="co">#&gt;  [70] tools_4.4.3              vipor_0.4.7              beeswarm_0.4.0          </span></span>
<span><span class="co">#&gt;  [73] httpuv_1.6.15            glue_1.8.0               restfulr_0.0.15         </span></span>
<span><span class="co">#&gt;  [76] promises_1.3.2           grid_4.4.3               generics_0.1.3          </span></span>
<span><span class="co">#&gt;  [79] gtable_0.3.6             tidyr_1.3.1              data.table_1.17.0       </span></span>
<span><span class="co">#&gt;  [82] BiocSingular_1.22.0      ScaledMatrix_1.14.0      XVector_0.46.0          </span></span>
<span><span class="co">#&gt;  [85] stringr_1.5.1            ggrepel_0.9.6            BiocVersion_3.20.0      </span></span>
<span><span class="co">#&gt;  [88] foreach_1.5.2            pillar_1.10.1            spam_2.11-1             </span></span>
<span><span class="co">#&gt;  [91] limma_3.62.2             later_1.4.1              benchmarkmeData_1.0.4   </span></span>
<span><span class="co">#&gt;  [94] dplyr_1.1.4              BiocFileCache_2.14.0     lattice_0.22-6          </span></span>
<span><span class="co">#&gt;  [97] rtracklayer_1.66.0       bit_4.6.0                tidyselect_1.2.1        </span></span>
<span><span class="co">#&gt; [100] locfit_1.5-9.12          scuttle_1.16.0           Biostrings_2.74.1       </span></span>
<span><span class="co">#&gt; [103] knitr_1.50               gridExtra_2.3            bookdown_0.42           </span></span>
<span><span class="co">#&gt; [106] edgeR_4.4.2              xfun_0.51                statmod_1.5.0           </span></span>
<span><span class="co">#&gt; [109] DT_0.33                  stringi_1.8.7            UCSC.utils_1.2.0        </span></span>
<span><span class="co">#&gt; [112] lazyeval_0.2.2           yaml_2.3.10              shinyWidgets_0.9.0      </span></span>
<span><span class="co">#&gt; [115] evaluate_1.0.3           codetools_0.2-20         tibble_3.2.1            </span></span>
<span><span class="co">#&gt; [118] BiocManager_1.30.25      cli_3.6.4                xtable_1.8-4            </span></span>
<span><span class="co">#&gt; [121] systemfonts_1.2.1        munsell_0.5.1            jquerylib_0.1.4         </span></span>
<span><span class="co">#&gt; [124] golem_0.5.1              Rcpp_1.0.14              dbplyr_2.5.0            </span></span>
<span><span class="co">#&gt; [127] scry_1.18.0              png_0.1-8                XML_3.99-0.18           </span></span>
<span><span class="co">#&gt; [130] parallel_4.4.3           pkgdown_2.1.1            ggplot2_3.5.1           </span></span>
<span><span class="co">#&gt; [133] blob_1.2.4               dotCall64_1.2            bitops_1.0-9            </span></span>
<span><span class="co">#&gt; [136] viridisLite_0.4.2        scales_1.3.0             purrr_1.0.4             </span></span>
<span><span class="co">#&gt; [139] crayon_1.5.3             rlang_1.1.5              KEGGREST_1.46.0</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Christine Hou, Jacqui Thompson, Stephanie C. Hicks.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
